nix/overlay.nix

```diff
@@ -0,0 +1,26 @@
+{ sources ? import ./sources.nix }:
+self: super:
+let
+  # Overrides all haskell packages to not automatically add cost centers.
+  # Note: profilingDetail is ends up as the `--profiling-detail` cabal flag,
+  # and as the `-fno-prof-auto` ghc flag. In nixpkgs this is set to
+  # "exported-functions" by default
+  disableAutoProfs = haskellSelf: haskellSuper: {
+    mkDerivation = drv:
+      haskellSuper.mkDerivation (drv // { profilingDetail = "none"; });
+  };
+
+  haskellOverlay = import ./haskell-overlay.nix {
+    inherit sources;
+    pkgs = self;
+  };
+in {
+  sources = if super ? sources then super.sources // sources else sources;
```

Comment:
Do we want to keep this step's `sources` attributes or the previous step's `sources`? `a // b` merges the two sets [prioritizing `a`](https://nixos.org/manual/nix/stable/language/operators.html#update) in case of key conflicts. This should not be an issue 99% of the cases, but it's such a tiny detail that if it does cause problems, it'll most likely take a while to figure out.

Reply:
Yeah it makes more sense the other way around, this is just what all of our other repos are doing so I did the same thing here. Probably worth going through everything to change this everywhere at some point.

Reply:
The docs say the former is taking precedence, but running this in a repl, that doesn't actually seem to be the case...

> Update [attribute set](https://nixos.org/manual/nix/stable/language/values.html#attribute-set) attrset1 with names and values from attrset2.

> The returned attribute set will have of all the attributes in e1 and e2. If an attribute name is present in both, the attribute value from the former is taken.

```
nix-repl> { a = 1; b = 2; }
{ a = 1; b = 2; }

nix-repl> { a = 1; b = 2; } // { a = 3; }
{ a = 3; b = 2; }
```

Reply:
Just putting it here for completeness as well, this seems to have been fixed in the docs https://github.com/NixOS/nix/commit/4ff9ed5c2de28e157d66b1a1ad3f24c18f6f9ee7, just not in the published version.

Reply:
Oh no

@alter2000 Could you revert all those PRs in the other repos? I can hereby pre-approve them, so you don't need to request a reviewer :sweat_smile:

Reply:
All handled, tested on Nix 2.11, .13, .15 and .16 and this seems to be the case. I'm curious what prompted the change in the docs though :thinking: